Userland/Libraries/LibJS/Lexer.cpp

```diff
@@ -24,7 +24,7 @@ HashMap<char, TokenType> Lexer::s_single_char_tokens;
 Lexer::Lexer(StringView source, StringView filename, size_t line_number, size_t line_column)
     : m_source(source)
     , m_current_token(TokenType::Eof, {}, {}, {}, filename, 0, 0, 0)
-    , m_filename(filename)
+    , m_filename(String::from_utf8(filename).release_value_but_fixme_should_propagate_errors())
```

Comment:
There is likely a better way - let me know what would be idiomatic here.

Reply:
You would have to make a factory function (the `static Class::create` ones) but that would mean Parser would also need one and so on.
So for now I think this is fine.